utilgo/tlsmanager/tlsmanager.go

```diff
@@ -173,9 +174,18 @@ func (m *tlsManager) WatchServerTLSConf(serverName ClientName, handler func(conf
 		case kclient.SecretChangeTypeAdd:
 			fallthrough
 		case kclient.SecretChangeTypeUpdate:
+			if m.tlsClientCertPEM[serverName] != nil && bytes.Compare(secret.Data[TLSSecretCrtName], m.tlsClientCertPEM[serverName]) == 0 {
+				return
+			}
+
+			m.tlsClientCertPEM[serverName] = secret.Data[TLSSecretCrtName]
 			config, err = m.generateTLSConfFromKeyPair(secret.Data[TLSSecretCrtName], secret.Data[TLSSecretKeyName])
+			m.tlsConf = config
```

Comment:
Doesn't this get set on line 199 anyway?

Reply:
Yeah, this one is redundant. Removed it.